tests/OperationProcessor.spec.ts

```diff
@@ -40,53 +48,230 @@ async function addBatchOfOneOperation (opBuf: Buffer, cas: Cas, transactionTime:
     batchFileHash: batchFileAddress
   };
 
-  const op = WriteOperation.create(opBuf, resolvedTransaction, 0);
+  const op = WriteOperation.create(opBuf, resolvedTransaction, operationIndex);
   return op;
 }
 
+function getHash (operation: WriteOperation): string {
+  const sha256HashCode = 18;
+
+  let contentBuffer;
+  if (operation.type === OperationType.Create) {
+    contentBuffer = Buffer.from(operation.encodedPayload);
+  } else {
+    contentBuffer = operation.operationBuffer;
+  }
+
+  const multihash = Multihash.hash(contentBuffer, sha256HashCode);
+  const multihashBase58 = Base58.encode(multihash);
+  return multihashBase58;
+}
+
+function getFactorial (n: number): number {
+  let factorial = 1;
+  for (let i = 2 ; i <= n ; ++i) {
+    factorial *= i;
+  }
+  return factorial;
+}
+
+// Return a permutation of a given size with a specified index among
+// all possible permutations. For example, there are 5! = 120 permutations
+// of size 5, so by passing index values 0..119 we can enumerate all
+// permutations
+function getPermutation (size: number, index: number): Array<number> {
+  let permutation: Array<number> = [];
```

Comment:
This is some crazy awesome implementation that I will not understand unless I spend a long time, so I will assume this works. Only nit is to use `const` for `permutation, j, t`.

Reply:
Done (const suggestion)